src/ec/suite_b/ecdsa/signing.rs

```diff
@@ -435,6 +454,44 @@ mod tests {
         );
     }
 
+    #[test]
```

Comment:
Tests of the public API should be done in tests/ecdsa_tests.rs.

I suggest that you add a constructor `pub(crate) try_from_test_vector` to `ring::digest::Digest` that accepts a digest algorithm and a precomputed value. Then you can rewrite the existing tests to use your new function instead of calling `sign_` directly as they currently do.